crates/top/rerun/src/commands/rrd/filter.rs

```diff
@@ -0,0 +1,168 @@
+                }
+
+                rrd_out.flush().context("couldn't flush output")?;
+
+                Ok(size_bytes)
+            });
+
+        for res in rx_decoder {
+            let mut is_success = true;
+
+            match res {
+                Ok(msg) => {
+                    let msg = match msg {
+                        re_log_types::LogMsg::ArrowMsg(store_id, mut msg) => {
+                            let (fields, columns): (Vec<_>, Vec<_>) =
+                                itertools::izip!(msg.schema.fields.iter(), msg.chunk.iter())
+                                    .filter(|(field, _col)| {
+                                        filter_timeline(&dropped_timelines, field)
+                                    })
+                                    .map(|(field, col)| (field.clone(), col.clone()))
+                                    .unzip();
+
+                            msg.schema.fields = fields;
+                            msg.chunk = re_log_types::external::arrow2::chunk::Chunk::new(columns);
+
+                            re_log_types::LogMsg::ArrowMsg(store_id, msg)
+                        }
+
+                        msg => msg,
+                    };
+
+                    tx_encoder.send(msg).ok();
+                }
+
+                Err(err) => {
+                    re_log::error!(err = re_error::format(err));
+                    is_success = false;
+                }
+            }
+
+            if !*best_effort && !is_success {
+                anyhow::bail!(
+                    "one or more IO and/or decoding failures in the input stream (check logs)"
+                )
+            }
+        }
+
+        std::mem::drop(tx_encoder);
+        let rrd_out_size = encoding_handle
+            .context("couldn't spawn IO thread")?
+            .join()
+            .unwrap()?;
+
+        let rrds_in_size = rx_size_bytes.recv().ok();
+        let filtered_ratio =
+            if let (Some(rrds_in_size), rrd_out_size) = (rrds_in_size, rrd_out_size) {
+                format!(
+                    "{:3.3}%",
+                    100.0 - rrd_out_size as f64 / (rrds_in_size as f64 + f64::EPSILON) * 100.0
+                )
+            } else {
+                "N/A".to_owned()
+            };
+
+        let file_size_to_string = |size: Option<u64>| {
+            size.map_or_else(
+                || "<unknown>".to_owned(),
+                |size| re_format::format_bytes(size as _),
+            )
+        };
+
+        re_log::info!(
+            dst_size_bytes = %file_size_to_string(Some(rrd_out_size)),
+            time = ?now.elapsed(),
+            filtered_ratio,
+            srcs = ?path_to_input_rrds,
+            srcs_size_bytes = %file_size_to_string(rrds_in_size),
+            "filter finished"
+        );
+
+        Ok(())
+    }
+}
+
+// ---
+
+use re_sdk::external::arrow2::datatypes::Field as ArrowField;
+
+fn filter_timeline(dropped_timelines: &HashSet<&String>, field: &ArrowField) -> bool {
```

Comment:
is more clear without context

```suggestion
fn keep_timeline(dropped_timelines: &HashSet<&String>, field: &ArrowField) -> bool {
```